src/modality-specific-files/task-events.md

```diff
@@ -177,8 +177,38 @@ The stimulus files can be added in a `/stimuli` directory
+
+{{ MACROS___make_columns_table(
+   {
+      "stim_file": ("REQUIRED", "There are no restrictions on the file formats of the stimuli files, but they should be stored in the `/stimuli` directory (under the root directory of the dataset; with OPTIONAL subdirectories). The values under the stim_file column correspond to a path relative to `/stimuli`. For example `images/cat03.jpg` will be translated to `/stimuli/images/cat03.jpg`. This column MUST appear as the first column in the file."),
+      "type": ("REQUIRED", "String. One of `movie`, `still_image`, `audio`, `annotation`, or Freeform.")
+   }
+) }}
+
+
+### Time-varying stimuli
+A single-line annotation in a `stimuli.tsv` column would be sufficient for the still stimulus files (for example, images). However, a time-varying stimulus file could need separate *temporal annotations* for every frame (that is, the smallest temporal resolution of the stimulus file).
+
+The temporal annotations for time-varying stimuli can be stored in a separate file, `stimulus.tsv`, which would hold the annotations for timepoints of the time-varying stimulus similar to an `events.tsv` file. `stimulus.tsv` MUST follow the same format and REQUIRED and OPTIONAL columns of the `events.tsv` file MAY be accompanied by a JSON file describing the columns in detail.
+
+An exmaplry `stimuli.tsv` file is shown below:
+
+| stim_file                            | type        | NSD_id | COCO_id | first_COCO_description                         | HED                                                                                                                                                                                            |
+| ------------------------------------ | ----------- | ------ | ------- | ---------------------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| nsd03050.png                         | still_image | 3050   | 262414  | a person standing on a surfboard riding a wave | (Foreground-view, ((Item-count/1, ((Human, Human-agent), Body, Male, Agent-trait/Adolescent)), (Play, (Item-count/1, Man-made-object)))), (Background-view, (Outdoors, Natural-feature/Ocean)) |
+| the_present_movie.mp4                | movie       | n/a    | n/a     | n/a                                            | Visual-presentation, Movie                                                                                                                                                                     |
+| the_present_stimulus-LogLumRatio.tsv | annotation  | n/a    | n/a     | n/a                                            | n/a                                                                                                                                                                                            |
+
```

Comment:
Does this mean that we should have something like this in JSON:

```json
{
  "COCO_id":{
    "Description": "ID of the image in the COCO database",
    "HED":"(Label/COCO_id, Categorical-value, Label/#)"
  }
}
```

Reference: HED Resources [BIDS JSON template](https://www.hed-resources.org/en/latest/BidsAnnotationQuickstart.html#create-a-json-template:~:text=JSON%20sidecar%20generated%20template.)

Reply:
Yes that is roughly the idea, but using the ID tag:

(ID/#, Label/COCO)

Under the Metadata tag we need to add an IRI tag in version 8.3.0.  If COCO IDS have an IRI, we could use that).  It could also include a URL for the COCO database in that group.